txpool/txpool.go

```diff
@@ -29,12 +29,21 @@ var (
 	ErrNegativeValue       = errors.New("negative value")
 	ErrNonEncryptedTxn     = errors.New("non-encrypted transaction")
 	ErrInvalidSender       = errors.New("invalid sender")
+	ErrUnderpriced         = errors.New("transaction underpriced")
 	ErrNonceTooLow         = errors.New("nonce too low")
 	ErrInsufficientFunds   = errors.New("insufficient funds for gas * price + value")
 	ErrInvalidAccountState = errors.New("invalid account state")
 	ErrAlreadyKnown        = errors.New("already known")
 )
 
+type TxOrigin = string
+
+const (
+	OriginAddTxn TxOrigin = "addTxn"
+	OriginReorg  TxOrigin = "reorg"
+	OriginGossip TxOrigin = "gossip"
+)
```

Comment:
Love that it's structured like this, instead of just having it baked into the code